src/CodeGen_LLVM.cpp

```diff
@@ -2404,11 +2404,19 @@ void CodeGen_LLVM::visit(const Call *op) {
         internal_assert(op->args.size() == 2);
         Value *a = codegen(op->args[0]);
         Value *b = codegen(op->args[1]);
+        if (op->args[0].type() != op->args[1].type()) {
+            llvm::Type *llvm_type_lhs = llvm_type_of(op->args[0].type());
+            b = builder->CreateBitCast(b, llvm_type_lhs);
```

Comment:
Is bitcast the right thing here? I would think we need `CreateZExtOrTrunc`.

Reply:
Both arguments have to be the same bit size, which should perhaps be asserted here. The LLVM IR docs say the LLVM types have to match even though the amount is *always* treated as an unsigned value, even if it is signed. So I think bitcast is the right thing, but there's probably some room for improvement.

Reply:
Hmm, I thought the plan was to allow shifts to have differently sized LHS and RHS types in the front and, and the only requirement was that the RHS be unsigned. Then we would normalize in the LLVM backend (since LLVM requires that) which would need `CreateZExtOrTrunc`, and any other backends that require this.

I see now that isn't the case in the front end (we need to be able to lossless_cast to an unsigned int of the size of the LHS type), but should it be?

Aside from that: LLVM types don't have signed-ness, so this cast will always be a no-op if the front-end doesn't allow differently sized LHS and RHS.

Reply:
The current design also looks good to me though.